rust-runtime/aws-smithy-xml/src/decode.rs

```diff
@@ -16,38 +16,64 @@ pub type Depth = usize;
 // much value in lots of different match variants
 
 #[derive(Debug)]
-pub enum XmlError {
+enum XmlDecodeErrorKind {
     InvalidXml(xmlparser::Error),
     InvalidEscape { esc: String },
     Custom(Cow<'static, str>),
     Unhandled(Box<dyn std::error::Error + Send + Sync + 'static>),
 }
 
-impl From<xmlparser::Error> for XmlError {
+#[derive(Debug)]
+pub struct XmlDecodeError {
+    kind: XmlDecodeErrorKind,
+}
+
+impl From<xmlparser::Error> for XmlDecodeError {
     fn from(error: xmlparser::Error) -> Self {
-        Self::InvalidXml(error)
+        Self {
+            kind: XmlDecodeErrorKind::InvalidXml(error),
+        }
     }
 }
 
-impl Display for XmlError {
+impl Display for XmlDecodeError {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        match self {
-            XmlError::InvalidXml(err) => write!(f, "XML parse error: {}", err),
-            XmlError::InvalidEscape { esc } => write!(f, "Invalid XML escape: {}", esc),
-            XmlError::Custom(msg) => write!(f, "Error parsing XML: {}", msg),
-            XmlError::Unhandled(err) => write!(f, "Error parsing XML: {}", err),
+        match &self.kind {
+            XmlDecodeErrorKind::InvalidXml(_) => write!(f, "XML parse error"),
+            XmlDecodeErrorKind::InvalidEscape { esc } => write!(f, "invalid XML escape: {}", esc),
+            XmlDecodeErrorKind::Custom(msg) => write!(f, "error parsing XML: {}", msg),
+            XmlDecodeErrorKind::Unhandled(_) => write!(f, "error parsing XML"),
```

Comment:
We should consider renaming this to `Unknown` to match our other enums. We are kind of handling it, so the current name is a bit odd.